lib/manager/npm/extract/monorepo.ts

```diff
@@ -13,10 +22,115 @@ function matchesAnyPattern(val: string, patterns: string[]): boolean {
   return res;
 }
 
-export function detectMonorepos(
+export async function extractPnpmFilters(
+  fileName: string
+): Promise<string[] | null> {
+  try {
+    const contents = yaml.safeLoad(await readLocalFile(fileName, 'utf8'), {
+      json: true,
+    }) as any;
+    if (
+      !Array.isArray(contents?.packages) ||
+      !contents.packages.every((item) => typeof item === 'string')
+    ) {
+      logger.debug(
+        { fileName },
+        'Failed to find required "packages" array in pnpm-workspace.yaml'
+      );
+      return null;
+    }
+    return contents.packages;
+  } catch (error) {
+    logger.debug({ fileName }, 'Failed to parse pnpm-workspace.yaml');
+    return null;
+  }
+}
+
+export async function findPnpmWorkspace(
+  packageFile: string,
+  normalizedLocalDir: string
+): Promise<{ lockFilePath: string; workspaceYamlPath: string } | null> {
+  const workspaceYamlPath = await findUpInsidePath(
```

Comment:
You shoud use `findLocalSiblingOrParent`

Reply:
resolved in 28a36aa130334e844a89884a6cb31c3fbf528659